bgpd/bgp_evpn.c

```diff
@@ -4333,6 +4333,64 @@ static void update_autort_vni(struct hash_bucket *bucket, struct bgp *bgp)
 	}
 }
 
+/*
+ * Handle autort change for L3VNI.
+ */
+static void update_autort_l3vni(struct bgp *bgp)
+{
+	bool updated = false;
+
+	if (!bgp->l3vni)
+		return;
+
+	if (!CHECK_FLAG(bgp->vrf_flags, BGP_VRF_IMPORT_RT_CFGD)) {
+		if (is_l3vni_live(bgp))
+			uninstall_routes_for_vrf(bgp);
+
+		/* Cleanup the RT to VRF mapping */
+		bgp_evpn_unmap_vrf_from_its_rts(bgp);
+
+		/* Remove auto generated RT */
+		evpn_auto_rt_import_delete_for_vrf(bgp);
+
+		list_delete_all_node(bgp->vrf_import_rtl);
+
+		/* Map auto derive or configured RTs */
+		evpn_auto_rt_import_add_for_vrf(bgp);
+
+		updated = true;
+	}
+
+	if (!CHECK_FLAG(bgp->vrf_flags, BGP_VRF_EXPORT_RT_CFGD)) {
+		list_delete_all_node(bgp->vrf_export_rtl);
+
+		evpn_auto_rt_export_delete_for_vrf(bgp);
+
+		evpn_auto_rt_export_add_for_vrf(bgp);
+
+		if (is_l3vni_live(bgp)) {
+			bgp_evpn_map_vrf_to_its_rts(bgp);
+			updated = true;
+		}
+	}
+	/* auto derive RD */
+	bgp_evpn_derive_auto_rd_for_vrf(bgp);
+
+	if (!is_l3vni_live(bgp))
+		return;
+
+	if (!updated)
```

Comment:
If you check if both RTs are manually configured and return early in the function, some of these if checks can be avoided/combined.

Reply:
ok. I will add this check early in the function, and remove updated usage.